clients/cli/src/prover.rs

```diff
@@ -157,22 +161,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         let start_time = Instant::now();
         let mut progress_time = start_time;
         for step in start..end {
-            proof = prove_seq_step(Some(proof), &pp, &tr).expect("error proving step");
+            // Generate proof for current step
+            proof =
+                prove_seq_step(Some(proof), &public_parameters, &tr).expect("error proving step");
             steps_proven += 1;
             completed_fraction = steps_proven as f32 / steps_to_prove as f32;
 
+            // Calculate performance metrics
             let progress_duration = progress_time.elapsed();
             let proof_cycles_hertz = k as f64 * 1000.0 / progress_duration.as_millis() as f64;
 
+            // Send progress update to orchestrator
             let progress = ClientProgramProofRequest {
                 steps_in_trace: total_steps as i32,
                 steps_proven,
                 step_to_start: start as i32,
-                program_id: "fast-fib".to_string(),
-                client_id_token: None,
-                proof_duration_millis: progress_duration.as_millis() as i32,
+                proof_duration_millis: progress_duration.as_millis() as i32, // TODO: find proof_duration_millis
                 k,
                 cli_prover_id: Some(prover_id.clone()),
+                network: 0,
```

Comment:
Should probably use the enum to get the value, if we're going to continue sending this. I think 0 is unknown.